pkg/cmd/secvulnMaven.go

```diff
@@ -5,33 +5,287 @@
 package cmd
 
 import (
+	"encoding/xml"
+	"errors"
 	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
 
 	"github.com/spf13/cobra"
 )
 
 var (
-    secvulnMavenCmd = &cobra.Command{
+	secvulnMavenCmd = &cobra.Command{
 		Use:   "maven",
 		Short: "Generate psuedo maven project for security vulnerability scanning",
 		Long:  "Generate psuedo maven project for security vulnerability scanning",
 		Run:   secvulnMavenExecute,
 	}
 
-	secvulnMavenParentDir     string
-	secvulnMavenPomUrls    *[]string
+	secvulnMavenParentDir string
+	secvulnMavenPomUrls   *[]string
+	secvulnMavenPomRepos  *[]string
+
+	completedProjects []Dependency
+	toDoProjects      []Dependency
 )
 
 func init() {
 	secvulnMavenCmd.PersistentFlags().StringVar(&secvulnMavenParentDir, "parent", "", "Parent project directory")
 	secvulnMavenPomUrls = secvulnMavenCmd.PersistentFlags().StringArray("pom", nil, "Component Pom URLs")
+	secvulnMavenPomRepos = secvulnMavenCmd.PersistentFlags().StringArray("repo", nil, "Repos to look for Poms")
 
 	secvulnMavenCmd.MarkPersistentFlagRequired("parent")
 	secvulnMavenCmd.MarkPersistentFlagRequired("pom")
+	secvulnMavenCmd.MarkPersistentFlagRequired("repo")
 
 	secvulnCmd.AddCommand(secvulnMavenCmd)
 }
 
 func secvulnMavenExecute(cmd *cobra.Command, args []string) {
 	fmt.Printf("Galasa Build - Security Vulnerability Maven - version %v\n", rootCmd.Version)
+
+	for _, pom := range *secvulnMavenPomUrls {
+		startScanningPom(pom)
+	}
+
+	updateParent()
+	fmt.Printf("Security scanning project pom.xml updated\n")
+
+}
+
+func startScanningPom(mainPomUrl string) {
+
+	fmt.Printf("Starting the scanning at main pom %v\n", mainPomUrl)
+
+	mainPom, err := readPomFromUrl(mainPomUrl)
+	if err != nil {
+		fmt.Printf("Unable to find the main pom at %v\n", mainPomUrl)
+		panic(err)
+	}
+
+	createPseudoMavenProject(mainPom)
+	fmt.Printf("Pseudo maven project created for: %v\n", mainPom.ArtifactId)
+
+	// Repeat the process for all projects in this dependency chain if the groupId is dev.galasa
+	fmt.Printf("Pseudo maven projects created for dependency chain of %v:\n", mainPom.ArtifactId)
+
+	for _, project := range toDoProjects {
+
+		// If this project has already been processed then continue
+		if checkIfCompleted(project) {
+			continue
+		}
+
+		// Get the current pom for this project from the repo to use to create the stripped down pom for the pseudo maven project
+		currentPom, err := readPomFromRepos(project.ArtifactId, project.GroupId, project.Version)
+
+		if err != nil || currentPom.ArtifactId != project.ArtifactId {
```

Comment:
do we need currentPom.ArtifactId != project.ArtifactId  ? either it reads it or it doesn't